docs/DOCUMENTATION_GUIDELINES.md

```diff
@@ -0,0 +1,227 @@
+
+The following is relevant to how to document parts of a crate that is a pallet.
+
+### Top Level Pallet Docs (`lib.rs`)
+
+for the top-level pallet docs, consider the following template:
+
+```
+//! # <Pallet Name>
+//!
+//! <single-liner about the pallet>.
+//!
+//! ## High-Level/End-User Details
+//!
+//! <a few paragraphs, focus on what external folks should know about the pallet>
+//!
+//! <The audience here is potentially non-coders who just want to know what this pallet does, not how it does it>
+//!
+//! ### Example
+//!
+//! <Your pallet must have a few tests that cover important user journeys. Use https://crates.io/crates/docify to reuse your these as examples>.
+//!
+//! ## Code Details
+//!
+//! <inside [`pallet`] module, a template is auto-generated that leads the reader to the relevant items. This is just a reminder to navigate to that page. Don't repeat things like "See Config trait for ..." that are generated inside [`pallet`] in here>
+//!
+//! See [`pallet`] module.
+//!
+//! <The audience of this is those who want to know how this pallet works, to the extent of being able to build something on top of it, like a DApp or another pallet>
+//!
+//! ## Low Level / Implementation Details
+//!
+//! <The format of this section is up to you, but we suggest the Design-oriented approach that follows>
+//!
+//! <The audience of this would be your future self, and those who want to deeply understand the pallet to optimize it further and such>
+//!
+//! ### Design Goals (optional)
+//!
+//! <Describe your goals with with the pallet design.>
+//!
+//! ### Design (optional)
+//!
+//! <Describe how you reach those goals. This should namely describe how your storage is laid out.>
+//!
+//! ### Terminology (optional)
+//!
+//! <Optionally, contain any custom terminology here. You can link to it if you want to use the terminology further up>
+```
+
+### Dispatchable
+
+For each dispatchable (`fn` item inside `pallet::call`), consider the following template:
+
+```
+/// <One liner explaining what this does>
+///
+/// ## Dispatch Origin
+///
+/// The dispatch origin of this call must be <details>
+///
+/// ## Details
+///
+/// <All other details. Focus on what errors could occur within this dispatch>
+///
+/// ## Errors (optional)
+///
+/// <If an extensive list of errors can be returned, list them individually instead of mentioning them in the section above>
+///
+/// ## Events (optional)
+///
+/// <Events are akin to the "return type" of dispatchables, optionally mention them>
+pub fn name(origin: OriginFor<T>, ...) -> DispatchResult {}
+```
+
+### Storage Items
+
+1. If a map-like type is being used, always note the choice of your hashers as private code docs (`// Hasher X chosen because ...`). Recall that this is not relevant information to external people, so it must be documented as `//`.
+2. Consider explaining why a storage type is always bounded.
```

Comment:
Isnt this enforced as long as there is no `without_storage_info`?

Reply:
I think this knowledge should be assumed and isn't relevant to mention here, unless there are design decisions around choosing a certain bound and what implications that could have for other developers using this pallet if they were to increase/decrease the bound (i.e. benchmarks will always be worst-case, so optimizations could be put in place based on a well known end-user base). This could go in the design section though. 

That said I'm curious what you had in mind @kianenigma - what would be an example of an explanation for "explaining why a storage type is always bounded" ?

Reply:
Okay, true, then I will flip it to: 

consider explaining why it is safe for the storage item to be unbounded, if `#[pallet::unbounded]` or `#[pallet::without_storage_info]` is being used.